### ISSUE TITLE
problem with absolute_path_linter()

### ISSUE THREAD

**Issue by FerandDalatieh** [jimhester/lintr#339]
I am using the development version 1.0.2.9000 and not able to figure out, how to use the function absolute_path_linter
https://github.com/jimhester/lintr/blob/baddc5b737cbdf8252ca9676b40cd6bb624426f0/R/path_linters.R#L181

### Here is what I am doing:
Create a file (code.R) which includes only the following assignments:

    file1 <- "/var"
    file2 <- "C:\System"
    file3 <- "~/docs"

save code.R and then use the following to detect lints:

    linters <- list(lintr::absolute_path_linter())
    lintr::lint("code.R", linters = linters)

I also tried setting parameter lax = FALSE, but that didn't make any difference.
### Actual behavior (output):
Only one lint is detected: '\S' is not a known escape sequence in line 2.
### Expected (wanted) behavior:
(Besides the lint about '\S') that all three assignments file1 - file3 should cause lints, because they all include absolute paths!

Documentation of lintr describes the following::
> absolute_path_linter: Check that no absolute paths are used (e.g.
> "/var", "C:\System", "~/docs").

Maybe I totally misunderstood what `absolute_path_linter()` should do, so any help or comment would be very helpful. Thank you!

Am using:
 - RStudio Version 1.1.383
 - R version 3.4.4

Side note: I asked the exact same question on [stackoverflow](https://stackoverflow.com/questions/51422586/lintr-how-to-check-that-no-absolute-paths-are-used) but got no answer.

**Comment by jimhester** [comment]
The code you have is not parsable, because of the `\S` escape, so no linters even get called. You need to double escape backslashes in R code, the line should be `file2 <- "C:\\System"`

**Comment by FerandDalatieh** [comment]
thank you @jimhester for the useful tip!
I used `file2 <- "C:\\System"` but now I don't get any markers when I use:
```
linters <- list(absolute_path_linter = lintr::absolute_path_linter())
lintr::lint("code.R", linters = linters)
```
Side note:
Using the following in last CRAN-version 1.0.2 works well and I got 3 markers for the three assignments, that there are absolute paths.
```
linters <- list(absolute_paths_linter = lintr::absolute_paths_linter)
lintr::lint("code.R", linters = linters)
```
